### PR TITLE
build kubevirt-testing image for multi-arch

### PR DESCRIPTION
**Pull request by zhlhahaha**
Build kubevirt-testing image for multi-arch

### CODE REVIEW

**Comment by zhlhahaha** [comment]
@davidvossel Can you review it when you have time?

**Review by dhiller — COMMENTED** [review]
/lgtm

Maybe we should add a presubmit job executing the build.sh? WDYT?

**Review by dhiller — APPROVED** [review]

**Comment by zhlhahaha** [comment]
> /lgtm
> 
> Maybe we should add a presubmit job executing the build.sh? WDYT?

Yes, and I do not find any prow job for building and publishing kubevirt-testing image. I am confused about how kubevirt-testing publish in dockerhub.

**Comment by zhlhahaha** [comment]
@phoracek Would you mind review this patch when you have time?

**Comment by rmohr** [comment]
/cc @fgimenez 

@fgimenez I think we will discuss all this also on kubevirt-dev and on the mailing list, FYI right now.

File: cluster-provision/images/kubevirt-testing/build.sh [hunk 1]
```diff
@@ -1,8 +1,19 @@
...
+ARCHS="amd64 arm64 ppc64le"
+
+# setup buildx environment
+docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
+docker buildx create --use
+
+# build multi-arch images
+for ARCH in ${ARCHS}; do
+        baseimage=$(grep "${ARCH}" ${SCRIPT_DIR}/BASEIMAGE | cut -d= -f2)
+        docker buildx build --load --platform linux/${ARCH}  -t kubevirtci/kubevirt-testing-${ARCH} --build-arg BASEIMAGE=${baseimage} ${SCRIPT_DIR}
```
Comment (fgimenez) [inline cluster-provision/images/kubevirt-testing/build.sh:18]:
```suggestion
        docker buildx build --load --platform linux/${ARCH}  -t quay.io/kubevirt/kubevirt-testing-${ARCH} --build-arg BASEIMAGE=${baseimage} ${SCRIPT_DIR}
```

File: cluster-provision/images/kubevirt-testing/publish.sh [hunk 2]
```diff
@@ -1,4 +1,17 @@
...
+for ARCH in ${ARCHS}; do
+        docker tag kubevirtci/kubevirt-testing-${ARCH} docker.io/${REPOSITORY}/kubevirt-testing-${ARCH}
+        docker push docker.io/${REPOSITORY}/kubevirt-testing-${ARCH}
+        manifest+=("docker.io/${REPOSITORY}/kubevirt-testing-${ARCH}:latest")
+done
+
+# create and push manifest
+docker manifest create --amend docker.io/${REPOSITORY}/kubevirt-testing:latest "${manifest[@]}"
+set -x; for ${ARCH} in ${ARCHS}; do docker manifest annotate --arch ${ARCH} docker.io/${REPOSITORY}/kubevirt-testing:latest docker.io/${REPOSITORY}/kubevirt-testing-${ARCH}:latest; done
+docker manifest push docker.io/${REPOSITORY}/kubevirt-testing:latest
```
Comment (fgimenez) [inline cluster-provision/images/kubevirt-testing/publish.sh:17]:
```suggestion
DOCKER_PREFIX=quay.io
REPOSITORY="kubevirt"
ARCHS="amd64 arm64 ppc64le"

# set tags and push multiarch images
for ARCH in ${ARCHS}; do
        docker tag ${DOCKER_PREFIX}/${REPOSITORY}/kubevirt-testing-${ARCH} ${DOCKER_PREFIX}/${REPOSITORY}/kubevirt-testing-${ARCH}
        docker push ${DOCKER_PREFIX}/${REPOSITORY}/kubevirt-testing-${ARCH}
        manifest+=("${DOCKER_PREFIX}/${REPOSITORY}/kubevirt-testing-${ARCH}:latest")
done

# create and push manifest
docker manifest create --amend ${DOCKER_PREFIX}/${REPOSITORY}/kubevirt-testing:latest "${manifest[@]}"
set -x; for ${ARCH} in ${ARCHS}; do docker manifest annotate --arch ${ARCH} ${DOCKER_PREFIX}/${REPOSITORY}/kubevirt-testing:latest ${DOCKER_PREFIX}/${REPOSITORY}/kubevirt-testing-${ARCH}:latest; done
docker manifest push ${DOCKER_PREFIX}/${REPOSITORY}/kubevirt-testing:latest
```

**Review by fgimenez — CHANGES_REQUESTED** [review]
Looks good, we only need to push the images to quay.io instead of docker.io, changes suggested for doing so. We need this https://github.com/kubevirt/project-infra/pull/789 to land first so that the required credentials are in place.

**Review by fgimenez — APPROVED** [review]
/lgtm
/approve
/hold

Looks good thanks! Let's wait until https://github.com/kubevirt/project-infra/pull/789 lands to merge this one.

**Comment by rmohr** [comment]
/close

With recent changes in kubevirt/kubevirt this PR is no longer needed to, since the test containers can be built for any kind of architecture now there. @zhlhahaha already incorporated these changes into this ppc64 PR in kubevirt/kubevirt.

**Pull request closed without merging**